(BUG) Compiler linking fatal error/crash using intrinsic fixed point procs when built/run w/ -debug, -o:none, or -o:minimal switches

### Hardware and Software
dev-2024-09 Odin Master Repo (release build)
LLVM 18.1.8
Windows 11 23H2 Intel 64bit
Build and Run both fail but switch sensitive, see below

### Symptoms
`odin build .` with these switches gives this
```
  -debug        linker crash
  -o:none       linker crash
  -o:minimal    linker crash    
  -o:size       no crash
  -o:speed      no crash
  -o:aggressive no crash
```
**odin code**
```go
package main
import "base:intrinsics"
import "core:fmt"
main :: proc() {
    x: i64 = 3223
    y: i64 = 4554
    j := intrinsics.fixed_point_mul(x, y, 2)
    fmt.printfln("%v",j)
    k := intrinsics.fixed_point_div(x, y, 2)
    fmt.printfln("%v",k)
}
```
**linking fatal error output**
```
a_dev-main-12926c5c180.obj : error LNK2019: unresolved external symbol __modti3 referenced in function main.main
a_dev-main-12926c5c180.obj : error LNK2019: unresolved external symbol __divti3 referenced in function main.main
fatal error LNK1120: 2 unresolved externals
```
### Expected behavior
Compiler does not crash when building these with the noted switches

### Recommendations
Beyond my level of knowledge to either debug or check source code for this type of linking issue

--- a/src/check_builtin.cpp
+++ b/src/check_builtin.cpp
@@ -5203,6 +5203,16 @@ gb_internal bool check_builtin_procedure(CheckerContext *c, Operand *operand, As
 				return false;
 			}
 
+			if (sz >= 64) {
+				if (is_type_unsigned(x.type)) {
+					add_package_dependency(c, "runtime", "umodti3", true);
+					add_package_dependency(c, "runtime", "udivti3", true);
+				} else {
+					add_package_dependency(c, "runtime", "modti3", true);
+					add_package_dependency(c, "runtime", "divti3", true);
+				}
+			}
+
 			operand->type = x.type;
 			operand->mode = Addressing_Value;
 		}

